config/main.py

```diff
@@ -2064,6 +2064,546 @@ def delete_snmptrap_server(ctx, ver):
+@click.argument('location', metavar='<location>', required=True, nargs=-1)
+@clicommon.pass_db
+def modify_location(db, location):
+    """ Modify snmp location"""
+    if isinstance(location, tuple):
+        location = " ".join(location)
+    elif isinstance(location, list):
+        location = " ".join(location)
+    snmp = db.cfgdb.get_table("SNMP")
+    try:
+        snmp_location = snmp['LOCATION']['Location']
+        if location in snmp_location:
+            click.echo("SNMP location {} already exists".format(location))
+            sys.exit(1)
+        else:
+            db.cfgdb.mod_entry('SNMP', 'LOCATION', {'Location': location})
+            click.echo("SNMP location {} modified in configuration".format(location))
+            try:
+                click.echo("Restarting SNMP service...")
+                clicommon.run_command("systemctl reset-failed snmp.service", display_cmd=False)
+                clicommon.run_command("systemctl restart snmp.service", display_cmd=False)
+            except SystemExit as e:
+                click.echo("Restart service snmp failed with error {}".format(e))
+                raise click.Abort()
+    except KeyError:
+        click.echo("Cannot modify SNMP Location.  You must use 'config snmp location add command <snmp_location>'")
+        sys.exit(2)
+
+
+from enum import IntEnum
+class SnmpUserError(IntEnum):
+    NameCheckFailure = 1
+    TypeNoAuthNoPrivOrAuthNoPrivOrPrivCheckFailure = 2
+    RoRwCheckFailure = 3
+    NoAuthNoPrivHasAuthType = 4
+    AuthTypeMd5OrShaOrHmacsha2IsMissing = 5
+    AuthTypeMd5OrShaOrHmacsha2Failure = 6
+    AuthPasswordMissing = 7
+    AuthPasswordFailsComplexityRequirements = 8
+    EncryptPasswordNotAllowedWithAuthNoPriv = 9
+    EncryptTypeDesOrAesIsMissing = 10
+    EncryptTypeFailsComplexityRequirements = 11
+    EncryptPasswordMissingFailure = 12
+    EncryptPasswordFailsComplexityRequirements = 13
+    UserAlreadyConfigured = 14
+
+
+@snmp.group(cls=clicommon.AbbreviationGroup)
+@clicommon.pass_db
+def user(db):
+    pass
+
+
+@user.command('add')
+@click.argument('user', metavar='<snmp_user>', required=True)
+@click.argument('user_type', metavar='<noAuthNoPriv|AuthNoPriv|Priv>', required=True)
+@click.argument('user_permission_type', metavar='<RO|RW>', required=True)
+@click.argument('user_auth_type', metavar='<MD5|SHA|HMAC-SHA-2>', required=False)
+@click.argument('user_auth_password', metavar='<auth_password>', required=False)
+@click.argument('user_encrypt_type', metavar='<DES|AES>', required=False)
+@click.argument('user_encrypt_password', metavar='<encrypt_password>', required=False)
+@clicommon.pass_db
+def add_user(db, user, user_type, user_permission_type, user_auth_type, user_auth_password, user_encrypt_type,
+             user_encrypt_password):
+    """ Add snmp user"""
+    if not snmp_username_check(user):
+        sys.exit(SnmpUserError.NameCheckFailure)
+    user_type = user_type.lower()
+    if not is_valid_user_type(user_type)[0]:
```

Comment:
>is_valid_user_type [](http://example.com/codeflow?start=11&length=18)

Better to call the function `is_valid_user_type` only once and store the return values into a tuple.

Reply:
updated.